shared/components/ember-output.jsx

```diff
@@ -0,0 +1,90 @@
+/* global React */
+
+export default React.createClass({
+    displayName: 'EmberOutput',
+
+    propTypes: {
+        source  : React.PropTypes.string.isRequired,
+        context : React.PropTypes.object.isRequired
+    },
+
+    injectMessages: function (locale, messages) {
+        var container  = this.container;
+        var lookupName = locale.toLowerCase();
+
+        if (container.has('locale:' + lookupName)) {
+            container.unregister('locale:' + lookupName);
+        }
+
+        container.register('locale:' + lookupName, {
+            locale:   locale,
+            messages: messages
+        }, {
+            instantiate: false
+        });
+    },
+
+    componentWillReceiveProps: function (nextProps) {
+        if (nextProps.source !== this.props.source) {
+            this.setState({});
+        }
+
+        var locale = Ember.get(nextProps, 'locales');
+
+        if (typeof locale === 'string') {
+            this.injectMessages(locale, nextProps.messages);
+            this.app.intl.set('locales', Ember.makeArray(locale));
+        }
+
+        this.controller.setProperties(nextProps);
+    },
+
+    componentWillUnmount: function () {
+        var app = this.app;
+
+        if (!app.isDestroying && !app.isDestroyed) {
+            app.destroy();
+        }
+    },
+
+    componentDidMount: function () {
+        var domElement = this.refs.placeholder.getDOMNode();
+        var locales    = Ember.makeArray(this.props.locales);
```

Comment:
We should make sure it can also work with a single String value, that's what the other FormatJS libs do (including `react-intl`), so they're aligned with the [`Intl` built-in APIs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat#Parameters).